src/base/memory_monitor_sr.hpp

```diff
@@ -0,0 +1,51 @@
+/*
+ *
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * memory_monitor_sr.hpp - Declaration of APIs and structures, classes
+ *                         for memory monitoring module
+ */
+
+#ifndef _MEMORY_MONITOR_SR_HPP_
+#define _MEMORY_MONITOR_SR_HPP_
+
+#include <string>
+
+namespace cubmem
+{
+  class memory_monitor
+  {
+    public:
+      memory_monitor (const char *server_name);
+      ~memory_monitor () {}
+
+      memory_monitor (const memory_monitor &) = delete;
+      memory_monitor (memory_monitor &&) = delete;
+
+      memory_monitor &operator = (const memory_monitor &) = delete;
+      memory_monitor &operator = (memory_monitor &&) = delete;
+
+    private:
+      static std::string make_tag_name (const char *file, const int line);
```

Comment:
Should make_tag_name() be static?

Reply:
It means that the function may not change member variable.

Reply:
I think you just add const keyword like `std::string make_tag_name (const char *file, const int line) const`. It meet your requirement.

Reply:
Actually, this function is not a member function(dependent on an instance), but a class function.

Reply:
In my point of view, static keyword allow us it can call without creating object. However this function is always called after creating memory_monitor object and it doesn't need to call without object. What is your opinion in this point?

Reply:
I change it static to const.

Reply:
As you said, this method may not be called if there is no instance, but it doesn't need any instance. I don't see an advantage to avoid calling it without an instance.

Reply:
hmm.. to me, the `static` keyword is exactly for this kind of method, and `const is not.

Reply:
Both opinion from you and me guarantee through private: and static keyword. I didn't consider private: keyword. So, I follow your suggestion.